src/wallet/wallet_tx_builder.cpp

```diff
@@ -0,0 +1,751 @@
+            },
+            [&](const UnifiedFullViewingKey& ufvk) {
+                // Transparent UTXOs will not have been selected if the UFVK does not contain
+                // a transparent key.
+                tfvk = ufvk.GetTransparentKey().value();
+            },
+            [&](const AccountZTXOPattern& acct) {
+                if (acct.GetAccountId() == ZCASH_LEGACY_ACCOUNT) {
+                    tfvk = pwalletMain->GetLegacyAccountKey().ToAccountPubKey();
+                } else {
+                    // By definition, we have a UFVK for every known account.
+                    auto ufvk = pwalletMain->GetUnifiedFullViewingKeyByAccount(acct.GetAccountId()).value();
+                    // Transparent UTXOs will not have been selected if the UFVK does not contain
+                    // a transparent key.
+                    tfvk = ufvk.GetTransparentKey().value();
+                }
+            },
+            [&](const auto& other) {
+                throw std::runtime_error("SelectOVKs: Selector cannot select transparent UTXOs.");
+            }
+        }, selector.GetPattern());
+        assert(tfvk.has_value());
+
+        auto ovks = tfvk.value().GetOVKsForShielding();
+        internalOVK = ovks.first;
+        externalOVK = ovks.second;
+    } else if (!spendable.sproutNoteEntries.empty()) {
+        // use the legacy transparent account OVKs when sending from Sprout
+        auto tfvk = pwalletMain->GetLegacyAccountKey().ToAccountPubKey();
+        auto ovks = tfvk.GetOVKsForShielding();
+        internalOVK = ovks.first;
+        externalOVK = ovks.second;
+    } else {
+        // This should be unreachable; it is left in place as a guard to ensure
+        // that when new input types are added to SpendableInputs in the future
+        // that we do not accidentally return the all-zeros OVK.
+        throw std::runtime_error("No spendable inputs.");
+    }
+
+    return std::make_pair(internalOVK, externalOVK);
+}
+
+PrivacyPolicy TransactionEffects::GetRequiredPrivacyPolicy() const
+{
+    PrivacyPolicy maxPrivacy = PrivacyPolicy::FullPrivacy;
+
+    if (!spendable.orchardNoteMetadata.empty() && payments.HasSaplingRecipient()) {
+        maxPrivacy = PrivacyPolicy::AllowRevealedAmounts;
+    }
+
+    if (!spendable.saplingNoteEntries.empty() && payments.HasOrchardRecipient()) {
+        maxPrivacy = PrivacyPolicy::AllowRevealedAmounts;
+    }
+
+    if (!spendable.sproutNoteEntries.empty() && payments.HasSaplingRecipient()) {
+        maxPrivacy = PrivacyPolicy::AllowRevealedAmounts;
+    }
+
+    bool hasTransparentSource = !spendable.utxos.empty();
+    if (payments.HasTransparentRecipient()) {
+        if (hasTransparentSource) {
+            // TODO: This is the correct policy, but it’s a breaking change from previous behavior,
+            //       so enable it separately.
+            // maxPrivacy = PrivacyPolicy::AllowFullyTransparent;
+            maxPrivacy = PrivacyPolicy::AllowRevealedSenders;
```

Comment:
Will this TODO be handled in a later PR? Or should it just be a separate commit in this PR?

Reply:
Yes, this is handled in #6409.